libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1921,6 +1947,15 @@ std::optional<SyncBlock> PbftManager::processSyncBlock() {
     return std::nullopt;
   }
 
+  // Check reward votes
+  auto reward_votes = vote_mgr_->updateRewardVotes(sync_block.first.pbft_blk->getPeriod() - 1);
```

Comment:
why ? you should be served sync blocks, which already contain reward votes maybe with 1 exception and that is if node has finalized period 10 and is processing(cert voting or whatever) block for period 11. If he serves you block 10 - it might not contain reward votes that will be added there after period 11 is finalized - how do you handle this ?

Reply:
If period 10 cert votes don't include all reward votes in period 11 block. Clear PBFT sync queue, mark peer as malicious players.

Here is an example for update reward votes in DB:
1. node receives enough cert votes and finalize PBFT block B1 with cert votes {v1, v2, v3}
2. node receive additional cert votes v4 for B1
3. node get partition (maybe lost connection with peers)
4. later node join back, find out of sync, then sync PBFT block B2 with reward votes {v1, v2, v3, v4}
5. node update v4 to B1 cert votes here

If reward votes queue include all reward votes objects, not just additional one that's not include in previous period block. I think that doesn't need update here